#23 [auction] 경매 도메인 설계

Kminss opened the pull request:
- 경매 도메인 (Auction.java) 엔티티 구현

## 개요 📝
<!---- 변경 사항 및 관련 이슈에 대해 간단하게 작성해주세요. 어떻게보다 무엇을 왜 수정했는지 설명해주세요. -->
- 경매 엔티티 구현 
<!---- Resolves: #(Isuue Number) -->
Resolves: #23 
 
## PR 유형 ✅
어떤 변경 사항이 있나요?

- [X] 새로운 기능 추가
- [ ] 버그 수정
- [ ] CSS 등 사용자 UI 디자인 변경
- [ ] 코드에 영향을 주지 않는 변경사항(오타 수정, 탭 사이즈 변경, 변수명 변경)
- [ ] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정
- [ ] 파일 혹은 폴더 삭제

## PR Checklist ✅
PR이 다음 요구 사항을 충족하는지 확인하세요.

- [X] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [ ] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

<br>

## 작업사항 🛠️
<br>

## 변경로직 ⚠️ 
<br>

### 변경전
<br>

### 변경후
<br>

## 기타

Kminss committed:
#23 [init-auction] 경매 도메인 설계

- 경매 도메인 (Auction.java) 엔티티 구현

Kminss committed:
Merge branch 'dev' into init/#23-auction-domain

Kminss committed:
#23 [init-auction] 경매 도메인 구현 - 입찰 엔티티 구현

- 입찰(Bid.java) 엔티티 구현

OuOHoon commented on src/main/java/com/sparta/ticketauction/domain/auction/entity/Auction.java:
```diff
@@ -1,4 +1,68 @@
+
+@Getter
+@NoArgsConstructor(access = AccessLevel.PROTECTED)
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
```
혹시 이 생성자는 언제 사용되는 건가용~? 아래에 private 생성자가 따로 있어서 궁금하네용~

Kminss replied:
기본 생성자와 전체 필드 생성자에 대해 외부에서 생성하는 것을 막는다는 것을 명시해두려고 작성했습니다.

OuOHoon replied:
> 기본 생성자와 전체 필드 생성자에 대해 외부에서 생성하는 것을 막기위해 걸어두었습니다.

제 생각엔 AllArgsConstructor를 제거하면 전체 필드 생성자는 외부에서 생성하지 못할 것 같은데 아닐까용~?

Kminss replied:
곰곰히 생각해보니 따로 명시해두지 않아도 코드에서 의도를 보이니 안해도 될 것같네요 수정하도록 하겠습니다~

Kminss replied:
[237457c](https://github.com/jshstar/ticket-auction/pull/42/commits/237457c7c63079e122d4499d6783bdc647d4945b)

수정했습니다

Kminss committed:
#23 [fix-auction] 경매 도메인 구현 - 경매 엔티티 컬럼 수정

- 시작일 -> 시작일시 startDateTime 으로 변경
- 마감일 -> 마감일시 endDateTime 으로 변경

Kminss committed:
#23 [fix-auction] 경매 도메인 구현 - 경매 엔티티 수정

- 생성자 수정
  of 메소드를 통해 Auction 엔티티를 생성할 때, 입찰가, 종료여부는 자동 초기화를 이용하여 생성하는 것으로 수정

- 시작일 -> 시작일시 startDateTime 으로 변경
- 마감일 -> 마감일시 endDateTime 으로 변경

jshstar commented on src/main/java/com/sparta/ticketauction/domain/auction/request/AuctionCreateRequest.java:
```diff
@@ -0,0 +1,24 @@
+	@NotBlank(message = "시작가를 입력해주세요.")
+	private final Long startPrice;
+
+	@JsonFormat(pattern = "yyyy-MM-dd HH")
```
@NotBlank 넣어주세요.

Kminss replied:
[7164d87](https://github.com/jshstar/ticket-auction/pull/42/commits/7164d873d954089a7d4bd15b85817dcb27fd71c4)

수정했습니다

jshstar replied:
고생하셨습니다~

Kminss committed:
#23 [fix-auction] 경매 도메인 구현 - 경매 추가 요청 DTO 수정

  - 경매 시작일시 @NotBlank 추가
  - 경매 마감일시 @NotBlank 추가

OuOHoon commented on src/main/java/com/sparta/ticketauction/domain/auction/entity/Bid.java:
```diff
@@ -1,4 +1,48 @@
+	private Long id;
+
+	@Comment("입찰한 유저")
+	@ManyToOne
```
혹시 즉시 로딩하는 이유가 무엇인지 알려주실 수 있나용~?

Kminss replied:
LAZY 로 수정하도록하겠습니다

Kminss replied:
[8a8a7d6](https://github.com/jshstar/ticket-auction/pull/42/commits/8a8a7d628c67d33f09d62a889807b06a5e9a0b6e)

수정했습니다.

OuOHoon reviewed (commented):
고생하셨어용~~

Kminss committed:
#23 [fix-auction] 경매 도메인 구현 - 경매 엔티티 수정

  - AllArgsConstructor 삭제
    해당 생성자에 대해 접근을 막는다는 것을 명시해두었는데
    현재 코드에서 그러한 의도가 보이기 떄문에 따로 명시하지 않아도 될 것같아
    삭제했습니다.

Kminss committed:
#23 [fix-auction] 경매 도메인 구현 - 입찰 엔티티 수정

  - ManyToOne fetch 타입 수정
    기본 EAGER 에서 LAZY 로 수정
